liferay-frontend/liferay-portal
# HR_209 If the sidebar menu is a link, the page title should change after activating the link

Reference: https://issues.liferay.com/browse/LPS-169101

In this PR, we are:
 * Turning screen navigation tag into a page subtitle contributor, so that the screen navigation `entry` and `category` labels are added to the page subtitle using portal API. 
 * Using the same `-` character to separate page title and subtitle elements. This provides a nice hierarchical visual schema to present the page title in the browser
 * For the case where a page is being configured, in addition, a suffix "(Configuring)" is added to the title. This follows the existing mechanism we have for the page edition.
 
How to test this:

- Follow LPS instructions or just go to any page configuration (press cog icon near to page in the control menu)
- Navigate through sections 
- Assert page title contains screen navigation entries and categories separated by dashes, before the original page title. For this case, you should see `(Configuring)` at the end of the page title
- For other cases, like object definition admin, just assert page title contains screen navigation entries and categories separated by dashes 

Other notes:
 * This solution does not guarantee screen navigation `entry` and `category` will show up in the page title as some widget might take over the responsibility of setting the full page title (e.g. user management)
 * Some screen navigation categories and entries are not well defined, so resulting page titles might look repetitive as both category and entry have the same value. This is technically not an issue in the page title composition mechanism but rather in the way screen navigation categories and entries are named/translated. That shall be addressed in separate tickets.
 * Page translation is another case where we'd like to apply the page title pattern, but I've left it out till we validate this

cc @ethib137 @marcoscv-work @edalgrin
